porch/func/internal/builtinevaluator.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/apply-replacements/replacements"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/starlark/starlark"
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	pb "github.com/GoogleContainerTools/kpt/porch/func/evaluator"
+	"k8s.io/klog/v2"
+)
+
+type builtinEvaluator struct {
+	pb.UnimplementedFunctionEvaluatorServer
+
+	fnMapping map[string]fn.ResourceListProcessor
+}
+
+func newBuiltInEvaluator() (*builtinEvaluator, error) {
+	return &builtinEvaluator{
+		fnMapping: map[string]fn.ResourceListProcessor{
+			"gcr.io/kpt-fn/starlark:v0.3.0": &starlark.Processor{},
```

Comment:
Something to think about (I don't have a good answer for it yet) is to what extent, especially for built-ins (where we can keep fairly high back compatibility bar) we can use the same compiled-in function for whole range of versions...

This is great for now though.